src/auth/auth.controller.ts

```diff
@@ -55,4 +56,34 @@ export class AuthController {
 
     return this.authService.createTokenResponse(sub, email, sessionId);
   }
+
+  @Post('password-reset-request')
+  async passwordResetRequest(
+    @Body() passwordResetRequestDto: PasswordResetRequestDto,
+  ): Promise<void> {
+    try {
+      const { user, token } =
+        await this.userService.createPasswordResetTokenForUser(
+          passwordResetRequestDto,
+        );
+
+      await this.notificationService.sendPasswordResetRequestMessage(
+        token,
+        user,
+      );
+    } catch (err) {
+      // Only throw errors other than NotFoundException to avoid exposing user
+      // registration details.
+      if (!(err instanceof NotFoundException)) {
+        throw err;
+      }
+    }
+  }
+
+  @Post('password-reset-set')
+  async passwordResetSet(
```

Comment:
It's kind of a strange name. 😅 Why not call it resetPassword?

Reply:
@Saela I debated long about this - the reason being that I feel that only `resetPassword` is misleading as well because this is the process which consists of two steps. Let's discuss this tomorrow :)